cmd/soci/commands/internal/platform.go

```diff
@@ -0,0 +1,70 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package internal
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/platforms"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/urfave/cli"
+)
+
+const (
+	PlatformFlagKey     = "platform"
+	AllPlatformsFlagKey = "all-platforms"
```

Comment:
would be good to have tests with create/push/pull functionality, which cover "all-platforms" flag

Reply:
should we do this in a future PR and move forward?

Reply:
I'd like for this to be a fast follow. There are a number of helper methods that assume there will be exactly 1 index so I think refactoring those to allow for the all-platforms flag will propagate across the tests and it would be clearer in a separate PR.

Reply:
I'm fine with that. Can you please create a ticket on that?

Reply:
Updated https://github.com/awslabs/soci-snapshotter/issues/54